src/operations/transforms/rotate.js

```diff
@@ -0,0 +1,84 @@
+const flatten = require('../utils/flatten')
+
+const mat4 = require('../../math/mat4')
+
+const {geom2, geom3, path2} = require('../../geometry')
+
+// Tait-Bryan Euler angle convention using active, intrinsic rotations around the axes in the order z-y-x
```

Comment:
huh, that is a new one for me :)

Reply:
ie, I do not know what it means :)

Reply:
Here's the theorem... https://en.wikipedia.org/wiki/Euler's_rotation_theorem

Here's the paper (very good reading) that I used for the matrix logic... http://danceswithcode.net/engineeringnotes/rotations_in_3d/rotations_in_3d_part1.html